src/main/java/io/confluent/connect/jdbc/TimestampIncrementingTableQuerier.java

```diff
@@ -59,25 +59,31 @@
   private String timestampColumn;
   private Long timestampOffset;
   private String incrementingColumn;
+  private boolean incrementingColumnUsePrimaryKey;
   private Long incrementingOffset = null;
   private long timestampDelay;
 
-  public TimestampIncrementingTableQuerier(QueryMode mode, String name, String topicPrefix,
+  public TimestampIncrementingTableQuerier(QueryMode mode, String name, String topicPrefix, String keyColumnName,
                                            String timestampColumn, Long timestampOffset,
-                                           String incrementingColumn, Long incrementingOffset, Long timestampDelay) {
-    super(mode, name, topicPrefix);
+                                           String incrementingColumn, boolean incrementingColumnUsePrimaryKey, Long incrementingOffset, Long timestampDelay) {
+    super(mode, name, topicPrefix, keyColumnName);
     this.timestampColumn = timestampColumn;
     this.timestampOffset = timestampOffset;
-    this.incrementingColumn = incrementingColumn;
     this.incrementingOffset = incrementingOffset;
     this.timestampDelay = timestampDelay;
+    this.incrementingColumn = incrementingColumn;
+    this.incrementingColumnUsePrimaryKey = incrementingColumnUsePrimaryKey;
   }
 
   @Override
   protected void createPreparedStatement(Connection db) throws SQLException {
-    // Default when unspecified uses an autoincrementing column
+    // Default when unspecified uses an autoincrementing column or primary key if primary key assumed to autoincrement.
     if (incrementingColumn != null && incrementingColumn.isEmpty()) {
-      incrementingColumn = JdbcUtils.getAutoincrementColumn(db, name);
+      if (incrementingColumn.isEmpty() && incrementingColumnUsePrimaryKey && keyColumn != null) {
+        incrementingColumn = keyColumn;
```

Comment:
Unlike auto increment columns, primary key columns aren't actually guaranteed to have a compatible type for this use, right? Should we be doing anything to check the type of the column? (Although I suppose it might have to wait until we actually get the relevant schema back)